### PR TITLE
Purchasability validation

### PR DESCRIPTION
**Pull request by tcichowicz**

### CODE REVIEW

File: test/PurchasabilityValidator.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      const actualCodes = await sut.isPurchasable(drugItem.address);
+      // then
+      expectPurchasabilityCodes(actualCodes).toEqual([purchasabilityCodes.TemperatureTooHigh]);
+    });
+  });
+
+  it('should return error code if temperature dropped below -22 degrees', async () => {
+    // given
+    const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, carrier1.id, carrier1.category);
+    const carrier = carrierWithTemperature(carrierCategories.Ship, -23);
```
Comment (przemyslawjanpietrzak) [inline test/PurchasabilityValidator.test.js:98]: magic number in variable will be less confusing

File: test/common/participants.js [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-const randomTemperature = () => Math.floor(-20 * Math.random());
-
 const participant = (id, category) => ({
   id,
   category,
 });
 
 const conditions = (category = carrierCategories.Truck) => ({
-  temperature: randomTemperature(),
+  temperature: -19,
```
Comment (przemyslawjanpietrzak) [inline test/common/participants.js]: as above

File: test/PurchasabilityValidator.test.js (hunk 1 above)
Comment (tcichowicz) [reply test/PurchasabilityValidator.test.js:98]: I'm not so sure about that - you see a `carrierWithTemperature()` with plainly a carrier category and the temperature. Additionally you have the context of the test name. Extracting a variable such as `temperatureBelowLimit` would add a bit more noise. But that's my opinion, was your perception of the code different from mine?

File: contracts/PurchasabilityValidator.sol [hunk 3]
```diff
@@ -1,13 +1,76 @@
...
 
     uint8 public VALID_FOR_PURCHASE = 100;
+    uint8 public NOT_IN_PHARMACY = 200;
+    uint8 public TOO_MANY_HANDOVERS = 201;
+    uint8 public TEMPERATURE_TOO_HIGH = 202;
+    uint8 public TEMPERATURE_TOO_LOW = 203;
+    uint8 public TOTAL_TRANSIT_TIME_TOO_LONG = 204;
+    uint8 public SINGLE_TRANSIT_TIME_TOO_LONG = 205;
 
     function isPurchasable(DrugItem _drugItem) public view returns (uint8[] memory result) {
```
Comment (kwedrowicz) [inline contracts/PurchasabilityValidator.sol:16]: This method is definitely too long and hard to read split to smaller private methods

File: test/PurchasabilityValidator.test.js (hunk 1 above)
Comment (przemyslawjanpietrzak) [reply test/PurchasabilityValidator.test.js:98]: In my opinion this `carrierWithTemperature(carrierCategories.Ship, temperatureBelowLimit);` looks more readable than `carrierWithTemperature(carrierCategories.Ship, -23);` but is not such important thing.

**Review by przemyslawjanpietrzak — APPROVED** [review]

**Review by kwedrowicz — APPROVED** [review]

**Pull request merged**